### PR TITLE
Feature/toggle gutter markers 2042

### LINKED ISSUE #2082: (design)Icons for left gutters and write mode
**Summary**
We need icons for the left gutter where the icons and spans for the blocks are shown.

They need to be toggle-able and located in the right area.

Image

### PR DESCRIPTION
**Pull request by mike-parkin**
**details**

Added toggle switch for the gutter markers extension to the source view,

- the switch is in the bottom toolbar.
- It will be disabled when the 'writeonly' mode is on
- added custom icon for this (and the writeonly mode as well)

Have fixed the glitch of the gutters, with the code/math style blocks,

Still to be resolved: major gutter glitches with QuoteBlocks, still some minor glitches with ListBlock, Some files not loading gutters in editor until user interacts with editor.

**Related Issues**:  #2042 #2043  closes #2082

### CODE REVIEW

File: web/src/views/source.ts [hunk 1]
```diff
@@ -91,6 +91,13 @@ export class SourceView extends TWLitElement {
   @property({ attribute: 'write-only', type: Boolean })
   writeOnly: boolean = false
 
+  /**
+   *  Turn on/off the node gutter markers.
+   *  Gutters will be disabled automatically in the "wite-only" mode.
```
Comment (nokome) [inline web/src/views/source.ts:96]:
```suggestion
   *  Gutters will be disabled automatically in "write-only" mode.
```

**Review by nokome — APPROVED** [review]

File: web/src/views/source/bottomPanel.ts [hunk 2]
```diff
@@ -113,6 +110,28 @@ class PanelElement extends LitElement {
     `
   }
 
+  private renderFormatSelect = () => {
+    const changeEvent = (e: Event) =>
+      (this.sourceView.format = (e.target as HTMLSelectElement).value)
```
Comment (simonwinter) [inline web/src/views/source/bottomPanel.ts:115]: Entirely an opinion: does anyone like the parenthesis style function body? For readability I believe that block style functions offer more clarity.

**Review by simonwinter — CHANGES_REQUESTED** [review]
Minor comment to review. Nothing wrong as such - just an opinion offered & perhaps if we feel it necessary, add a rule to linting.

File: web/src/views/source/bottomPanel.ts (hunk 2 above)
Comment (nokome) [reply web/src/views/source/bottomPanel.ts:115]: I'm not a fan of the readability of arrow functions generally but agree the block style is slightly less worse than parentheses.

**Pull request merged**